Exiled.Events/Patches/Events/Player/ReservedSlot.cs

```diff
@@ -0,0 +1,111 @@
+// -----------------------------------------------------------------------
+// <copyright file="PreAuthenticating.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Reflection;
+using Exiled.API.Enums;
+using GameCore;
+using Mirror.LiteNetLib4Mirror;
+using PlayerRoles.PlayableScps.Scp079;
+using PluginAPI.Events;
+using Log = Exiled.API.Features.Log;
+
+namespace Exiled.Events.Patches.Events.Player
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.Events.EventArgs.Player;
+    using Handlers;
+
+    using HarmonyLib;
+
+    using LiteNetLib;
+    using LiteNetLib.Utils;
+
+    using NorthwoodLib.Pools;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="ReservedSlot.HasReservedSlot(string userId, out bool bypass)" />.
+    ///     Adds the <see cref="Player.ReservedSlot" /> event.
+    /// </summary>
+    [HarmonyPatch(typeof(ReservedSlot), "HasReservedSlot")]
+    internal static class ReservedSlotPatch
+    {
+        [HarmonyTranspiler]
+        private static IEnumerable<CodeInstruction> ReservedSlotPatchTranspiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            LocalBuilder jumpConditions = generator.DeclareLocal(typeof(ReservedSlotEventResult));
+
+            Label continueConditions = generator.DefineLabel();
+
+            Label allowUnconditional = generator.DefineLabel();
+            Label returnTrue = generator.DefineLabel();
+            Label returnFalse = generator.DefineLabel();
+
+
+            int offset = -1;
+            int index = newInstructions.FindLastIndex(
+                instruction => instruction.LoadsField(Field(typeof(PlayerCheckReservedSlotCancellationData), nameof(PlayerCheckReservedSlotCancellationData.HasReservedSlot)))) + offset;
+
+            newInstructions[index].WithLabels(continueConditions);
+
+            newInstructions.InsertRange(
+                index,
+                new[]
+                {
+                    new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
```

Comment:
Your comments below are fine, but you should add what those uncommented instructions do like in the other transpilers, they could be useful for new contributors.